src/main/java/com/firebolt/jdbc/service/FireboltStatementService.java

```diff
@@ -33,17 +33,19 @@ public class FireboltStatementService {
 	 * @param properties the connection properties
 	 * @param queryTimeout         query timeout
 	 * @param maxRows              max rows
+	 * @param maxFieldSize         max field size
 	 * @param standardSql          indicates if standard sql should be used
 	 * @param systemEngine         indicates if system engine is used
 	 * @param statement           the statement
 	 * @return an InputStream with the result
 	 */
+	@SuppressWarnings("java:S107") //Number of parameters (8) > max (7). Not nice but probably not so bad for internal class
 	public Optional<ResultSet> execute(StatementInfoWrapper statementInfoWrapper,
-									   FireboltProperties properties, int queryTimeout, int maxRows, boolean standardSql, boolean systemEngine, FireboltStatement statement)
+									   FireboltProperties properties, int queryTimeout, int maxRows, int maxFieldSize, boolean standardSql, boolean systemEngine, FireboltStatement statement)
```

Comment:
I feel like suppressing this warning everywhere is not great, even in internal classes. Having methods with many parameters can impact maintainability and readability of the code (see PK for an example of number of parameters getting out of hand).
In here, is it possible to create a struct/class `ExecuteProperties` and store `queryTimeout`, `maxRows`, `maxFieldSize`, `standardSql`, and possibly `systemEngine` flags there? Or use some other method of keeping parameters manageable.

Reply:
You are absolutely right. I have done this in separate PR because this change required a lot of changes in tests.